httpie/core.py

```diff
@@ -188,7 +188,7 @@ def request_body_read_callback(chunk: bytes):
             args.follow = True  # --download implies --follow.
             downloader = Downloader(output_file=args.output_file, progress_file=env.stderr, resume=args.download_resume)
             downloader.pre_request(args.headers)
-        messages = collect_messages(args=args, config_dir=env.config.directory,
+        messages = collect_messages(env, args=args, config_dir=env.config.directory,
```

Comment:
Here and elsewhere: Could we be more specific in what the functions need from `env`? If it would be too cumbersome, then we could probably remove the specific args like `config_dir` here.

Reply:
> Here and elsewhere: Could we be more specific in what the functions need from env? If it would be too cumbersome, then we could probably remove the specific args like config_dir here.

Sure thing. I did not notice `config_dir` was simply an attribute of `env`.

Reply:
For the record, there’s a strong preference for being specific because, without it, it’s really hard to understand what the function tree depends on, which makes reasoning about many things harder.